pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -780,12 +781,26 @@ func (vs *VSphere) DetachDisk(volPath string, nodeName k8stypes.NodeName) error
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
 		vsi, err := vs.getVSphereInstance(nodeName)
-		if err != nil {
-			// If node doesn't exist, disk is already detached from node.
-			if err == vclib.ErrNoVMFound {
-				glog.Infof("Node %q does not exist, disk %s is already detached from node.", convertToString(nodeName), volPath)
+		if err == vclib.ErrNoVMFound {
+			// If node doesn't exist, discover the node to check if node is still in vcenter inventory
+			node, err := vs.nodeManager.GetRemovedNode(nodeName)
```

Comment:
I think this will not work if controller-manager restarts since this information will vanish. Can you please confirm this?